pulp_ansible/app/tasks/collections.py

```diff
@@ -71,22 +71,52 @@ def sync(remote_pk, repository_pk, mirror):
     d_version.create()
 
 
-def import_collection(artifact_pk, repository_pk=None):
+def import_collection(
+    artifact_pk,
+    repository_pk=None,
+    expected_namespace=None,
+    expected_name=None,
+    expected_version=None,
+):
     """
-    Create a Collection from an uploaded artifact.
+    Create a Collection from an uploaded artifact and optionally validate its expected metadata.
+
+    This task provides optional validation of the `namesapce`, `name`, and `version` metadata
+    attributes. If the Artifact fails validation, the Artifact is deleted and the Collection is not
+    created.
 
     Args:
-        artifact_pk (str): The pk or the Artifact to create the Collection from.
+        artifact_pk (str): The pk of the Artifact to create the Collection from.
 
     Keyword Args:
         repository_pk (str): Optional. If specified, a new RepositoryVersion will be created for the
             Repository and any new Collection content associated with it.
+        expected_namespace (str): Optional. The namespace is validated against the namespace
+            specified in the Collection's metadata. If it does not match an AssertionError is
+            raised.
+        expected_name (str): Optional. The name is validated against the name specified in the
+            Collection's metadata. If it does not match an AssertionError is raised.
+        expected_version (str): Optional. The version is validated against the version specified in
+            the Collection's metadata. If it does not match an AssertionError is raised.
     """
     artifact = Artifact.objects.get(pk=artifact_pk)
     log.info("Processing collection from {path}".format(path=artifact.file.path))
     importer_result = process_collection(str(artifact.file.path))
     collection_info = importer_result["metadata"]
 
+    try:
+        if expected_namespace:
+            assert expected_namespace is collection_info["namespace"]
+
+        if expected_name:
+            assert expected_name is collection_info["name"]
+
+        if expected_version:
+            assert expected_version is collection_info["version"]
+    except AssertionError:
+        artifact.delete()
```

Comment:
Artifact is deleted if it fails validation against expected data, but it's not deleted if process_collection fails.

Reply:
Agreed that case should cause the Artifact to delete also. Will fix now, ty.

Reply:
fixed in next push.